datafusion/functions/src/core/map.rs

```diff
@@ -28,7 +28,22 @@ use datafusion_common::{exec_err, internal_err, ScalarValue};
 use datafusion_common::{not_impl_err, Result};
 use datafusion_expr::{ColumnarValue, ScalarUDFImpl, Signature, Volatility};
 
+/// Check if we can evaluate the expr to constant directly.
+///
+/// # Example
+/// ```sql
+/// SELECT make_map('type', 'test') from test
+/// ```
+/// We can evaluate the result of `make_map` directly.
+#[inline]
```

Comment:
Any specific reason for using inline over here ?

Reply:
Just a personal habit to mark the very short function.
Will it cause some problems that I haven't noticed before?

Reply:
Since it more of a suggestion to complier, i dont think this should cause problems. TIL: Inlining small functions can provide small speed wins.

Reply:
I think there is rare case for `inline`. I perfer not to inline unless the benmark shows that compiler fail to do the good job for us

https://matklad.github.io/2021/07/09/inline-in-rust.html

Reply:
Seem inline here is actually unnessary according to the article mentioned by @jayzhan211 , removed.